### DISCUSSION TITLE
Vertical scrollbars when putting a `Vertical` layout into `Collapsible` instead of the widget directly

### DISCUSSION THREAD

**Discussion by kostrykin** [Textualize/textual#4917] [Q&A] [answered]
Consider this MWE:
```python
from textual.app import App
from textual.widgets import Collapsible, Label
from textual.containers import Vertical

class CollapsibleApp(App):

    def compose(self):
        with Collapsible(collapsed=False, title="Leto"):
            with Vertical():
                yield Label('line of text\n' * 20)

if __name__ == "__main__":
    app = CollapsibleApp()
    app.run()
```

The problem with it is that we don't get vertical scrollbars:
<img width="1277" alt="Bildschirmfoto 2024-08-21 um 20 55 00" src="https://github.com/user-attachments/assets/b7f06acb-7d82-4cd0-97dc-be653943521e">

We would get vertical scrollbars when putting the `Label` directly into the `Collapsible`. But how to make it work with the `Vertical` layout in between?

---

Why would one want to use a `Vertical` layout in between, you might ask? Because when mounting widgets directly into the `Collapsible` dynamically,
```python
class CollapsibleApp(App):

    BINDINGS = [
        ('x', 'add_widget', 'Add widget')
    ]

    def compose(self):
        with Collapsible(collapsed=False, title="Leto", id='collapsible'):
            yield Label('line of text')

    def action_add_widget(self):
        self.query_one('#collapsible').mount(Label('line of text 2'))
```
they get mounted _after_ the `Collapsible` instead, for some reason, instead of inside:
<img width="277" alt="Bildschirmfoto 2024-08-21 um 21 01 14" src="https://github.com/user-attachments/assets/244503bd-0ba5-428e-8dcd-b42c73a85b1b">

**Comment by willmcgugan** [comment, marked as answer]
You can swap your `Vertical` for a `VerticalScroll`.

As for your original issue, the Collapsible has children of its own for the title and contents. If you want to add something to the contents, I think the following should work:

```python
self.query_one('#collapsible > Contents').mount(Label('line of text 2'))
```

**Reply by TomJGooding** [reply]
Thanks Will, I've deleted my original reply. Just to note regarding the  `VerticalScroll`, it really depends on whether you want the entire screen to scroll or just the container itself.

**Reply by kostrykin** [reply]
Much thanks, another solution I have found meanwhile, will leave it here for the record:
```python
    def compose(self):
        with Collapsible(collapsed=False, title="Leto"):
            vert = Vertical()            # <---
            vert.styles.height = 'auto'  # <---
            with vert:                   # <---
                yield Label('line of text\n' * 20)
```
